File: psra/service/src/main/java/by/arhor/psra/exception/EntityNotFoundException.java
package by.arhor.psra.exception;

import by.arhor.psra.CoreVersion;
import by.arhor.psra.localization.Label;

public final class EntityNotFoundException extends FieldValueException {

	private static final long serialVersionUID = CoreVersion.SERIAL_VERSION_UID;

	public EntityNotFoundException(Label label, String fieldName, Object fieldValue) {
		super(label, fieldName, fieldValue);
	}
}

File: psra/service/src/main/scala/by/arhor/psra/exception/EntityNotFoundException.scala
package by.arhor.psra.exception

import by.arhor.psra.CoreVersion
import by.arhor.psra.localization.Label

import scala.beans.BeanProperty

object EntityNotFoundException {
	val serialVersionUID: Long = CoreVersion.SERIAL_VERSION_UID
}

final class EntityNotFoundException(
	@BeanProperty val label: Label,
	@BeanProperty val fieldName: String,
	@BeanProperty val fieldValue: AnyRef
) extends LocalizedException {

	override def getParams: Array[AnyRef] = Array(fieldName, fieldValue)

}
